evm/src/cpu/kernel/asm/util/basic_macros.asm

```diff
@@ -229,3 +229,70 @@
     %select_bool
     // stack: max
 %endmacro
+
+%macro u32
+    %and_const(0xffffffff)
+%endmacro
+
+%macro not_u32
+    // stack: x
+    PUSH 0xffffffff
+    // stack: 0xffffffff, x
+    SUB
+    // stack: 0xffffffff - x
+%endmacro
+
+%macro add3_u32
+    // stack: x, y, z
+    ADD
+    // stack: x+y, z
+    ADD
+    // stack: x+y+z
+    %u32
+%endmacro
+
+
+// given u32 bytestring abcd return dcba
+%macro reverse_bytes_u32
+    // stack: abcd
+    DUP1
+    %and_const(0xFF)
+    // stack: d, abcd
+    %stack (d, abcd) -> (abcd, d, 0x100, d)
+    // stack: abcd, d, 0x100, d
+    SUB
+    DIV
+    // stack: abc, d
+    DUP1
+    %and_const(0xFF)
+    // stack: c, abc, d
+    %stack (c, abc) -> (abc, c, 0x100, c)
+    // stack: abc, c, 0x100, c, d
+    SUB
+    DIV
+    // stack: ab, c, d
+    DUP1
+    %and_const(0xFF)
+    // stack: b, ab, c, d
+    %stack (b, ab) -> (ab, b, 0x100, b)
+    // stack: ab, b, 0x100, b, c, d
+    SUB
+    DIV
+    // stack: a, b, c, d
+    SWAP1
+    %shl_const(8)
+    OR
+    // stack: ba, c, d
+    SWAP1
+    %shl_const(16)
+    OR
+    // stack: cba, d
+    SWAP1
+    %shl_const(24)
+    OR
+    // stack: dcba
+%endmacro
+
+%macro break
```

Comment:
If this will be merged, let's call it something like `%debug` or `%deadbeef`, to make it more clear that it's not something that should be in any production code, and maybe add a note that this causes termination for debugging purposes.

Reply:
I think for debugging purposes we can also just add a `STOP` opcode? So maybe this isn't useful

Reply:
I got rid of it!